runelite-client/src/main/java/net/runelite/client/plugins/raids/bats/BatsOverlay.java

```diff
@@ -0,0 +1,100 @@
+package net.runelite.client.plugins.raids.bats;
+
+import com.google.inject.Inject;
+import java.awt.Color;
+import java.awt.Dimension;
+import java.awt.Graphics2D;
+import net.runelite.api.Client;
+import net.runelite.api.Perspective;
+import net.runelite.api.Player;
+import net.runelite.api.Point;
+import net.runelite.api.coords.LocalPoint;
+import net.runelite.api.coords.WorldPoint;
+import net.runelite.client.plugins.raids.RaidsConfig;
+import net.runelite.client.plugins.raids.RaidsPlugin;
+import net.runelite.client.ui.overlay.Overlay;
+import net.runelite.client.ui.overlay.OverlayLayer;
+import net.runelite.client.ui.overlay.OverlayPosition;
+import net.runelite.client.ui.overlay.OverlayUtil;
+import net.runelite.client.ui.overlay.components.ProgressPieComponent;
+
+public class BatsOverlay extends Overlay
+{
+	private RaidsPlugin plugin;
+	private RaidsConfig config;
+	private Client client;
+	private ProgressPieComponent pie = new ProgressPieComponent();
+
+	@Inject
+	public BatsOverlay(RaidsConfig config, RaidsPlugin plugin, Client client)
+	{
+		this.config = config;
+		this.plugin = plugin;
+		this.client = client;
+		setPosition(OverlayPosition.DYNAMIC);
+		setLayer(OverlayLayer.ABOVE_SCENE);
+		pie.setProgress(100);
+		pie.setBorderColor(Color.black);
+	}
+
+	@Override
+	public Dimension render(Graphics2D graphics)
+	{
+		BatsLocator batsLocator = plugin.getBatsLocator();
+		if (config.enableBatsLocator() && plugin.isInRaidChambers() && batsLocator.isDrawChestNumbers())
+		{
+			int[] chestCounts = batsLocator.getChestCounts();
+			int i = -1;
+			for (WorldPoint chestLocation : batsLocator.getChestLocations())
+			{
+				i++;
+				Chest chest = batsLocator.getChestMap().get(chestLocation);
+				if (chest.getState() == Chest.State.GRUBS)
+				{
+					continue;
+				}
+				Player player = client.getLocalPlayer();
+				if (player != null)
```

Comment:
This and the condition below do not change between iterations. Pull them out of the loop.